src/Calendar.jsx

```diff
@@ -153,7 +157,7 @@ let Calendar = React.createClass({
     //if the value changes reset views to the new one
     if ( !dates.eq(val, dateOrNull(this.props.value), VIEW_UNIT[view]))
       this.setState({
-        currentDate: val ? new Date(val) : new Date()
+        currentDate: this.getDateOrCurrent(val)
```

Comment:
That's what started to do.
But then many tests using `BaseCalendar` were failing because they couldn't find `this.props.onCurrentDateChange`. 
How to access the wrapped container state and methods if we replace `BaseCalendar` by `Calendar` ?